odxtools/minmaxlengthtype.py

```diff
@@ -34,113 +34,136 @@ def __post_init__(self):
+                is_highlow_byte_order=self.is_highlow_byte_order,
+            ))
+
+        # TODO: ensure that the termination delimiter is not
+        # encountered within the encoded value.
+
+        if encode_state.is_end_of_pdu or len(value_bytes) == self.max_length:
+            # All termination types may be ended by the end of the PDU
+            # or once reaching the maximum length. In this case, we
+            # must not add the termination sequence
+            pass
         else:
-            termination_char = self.__termination_character()
-            if self.termination == "END-OF-PDU":
-                termination_char = b''
-            odxassert(
-                termination_char is not None,
-                f"MinMaxLengthType with termination {self.termination}"
-                f"(min: {self.min_length}, max: {self.max_length}) failed encoding {internal_value}"
-            )
-            return value_byte + termination_char
+            termination_sequence = self.__termination_sequence()
+
+            # ensure that we don't try to encode an odd-length
+            # value when using a two-byte terminator
+            odxassert(len(value_bytes) % len(termination_sequence) == 0)
+
+            value_bytes.extend(termination_sequence)
+
+        if len(value_bytes) < self.min_length:
+            raise EncodeError(f"Encoded value for MinMaxLengthType "
+                              f"must be at least {self.min_length} bytes long. "
+                              f"(Is: {len(value_bytes)} bytes.)")
+        elif self.max_length is not None and len(value_bytes) > self.max_length:
+            raise EncodeError(f"Encoded value for MinMaxLengthType "
+                              f"must not be longer than {self.max_length} bytes. "
+                              f"(Is: {len(value_bytes)} bytes.)")
+
+        return value_bytes
 
     def convert_bytes_to_internal(self, decode_state: DecodeState, bit_position: int = 0):
         if decode_state.cursor_position + self.min_length > len(decode_state.coded_message):
-            raise DecodeError("The PDU ended before min length was reached.")
+            raise DecodeError("The PDU ended before minimum length was reached.")
 
         coded_message = decode_state.coded_message
-        cursor_position = decode_state.cursor_position
-        termination_char = self.__termination_character()
+        cursor_pos = decode_state.cursor_position
+        termination_seq = self.__termination_sequence()
 
-        # If no termination char is found, this is the next byte after the parameter.
-        max_termination_byte = len(coded_message)
+        max_terminator_pos = len(coded_message)
         if self.max_length is not None:
-            max_termination_byte = min(max_termination_byte, cursor_position + self.max_length)
+            max_terminator_pos = min(max_terminator_pos, cursor_pos + self.max_length)
 
         if self.termination != "END-OF-PDU":
```

Comment:
What's the default value for `self.termination` if not specified in the XML?

Reply:
the XSD says it is a required attribute...